### PR TITLE
adding lower and upper extraction fn

### PR DESCRIPTION
**Pull request by b-slim**
fix for issue #2010

### CODE REVIEW

File: processing/src/main/java/io/druid/query/extraction/LowerExtractionFn.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package io.druid.query.extraction;
```
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/query/extraction/LowerExtractionFn.java]: license header

**Comment by xvrl** [comment]
@b-slim upper and lowercase are locale sensitive, can we add an optional locale parameters? You can look at https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/extraction/TimeFormatExtractionFn.java#L45 for an example where we allow for locale.

**Comment by fjy** [comment]
@b-slim let's finish this one up

File: processing/src/main/java/io/druid/query/extraction/LowerExtractionFn.java (hunk 1 above)
Comment (b-slim) [inline processing/src/main/java/io/druid/query/extraction/LowerExtractionFn.java]: done

**Comment by b-slim** [comment]
@xvrl and @fjy this should be good to go

**Comment by b-slim** [comment]
@gianm does this fix your issue ?

File: processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java [hunk 2]
```diff
@@ -35,4 +35,6 @@
   public static final byte CACHE_TYPE_ID_SUBSTRING = 0x8;
   public static final byte CACHE_TYPE_ID_CASCADE = 0x9;
   public static final byte CACHE_TYPE_ID_STRING_FORMAT = 0xA;
+  public static final byte UPPER_EXTRACTION_FN_CACHE_ID = 0xB;
```
Comment (fjy) [inline processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java]: let's make the naming consistent

**Comment by fjy** [comment]
:+1: I have a minor comment

File: processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java (hunk 2 above)
Comment (himanshug) [inline processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java]: +1

**Comment by himanshug** [comment]
:+1: after https://github.com/druid-io/druid/pull/2076#discussion_r50343923 is resolved

File: processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java (hunk 2 above)
Comment (b-slim) [inline processing/src/main/java/io/druid/query/extraction/ExtractionCacheHelper.java]: done

**Comment by b-slim** [comment]
@fjy and @himanshug thanks ! will merge it once Travis is happy !

**Pull request merged**